style-guide.rst

```diff
@@ -0,0 +1,625 @@
+- Readers skim. Section titles should be clear and provide information.
+
+  
+.. _writing-about-code:
+  
+Writing code and writing about code
+======================================
+
+ODK Documentation includes code samples in a number of languages. Make sure to follow generally accepted coding style for each language. 
+
+.. _indenting:
+
+Indenting
+------------
+
+In code samples:
+
+- Use spaces, not tabs.
+- Two spaces for logical indents in most languages.
+
+  - Python samples must use `four spaces per indent level <https://www.python.org/dev/peps/pep-0008/#indentation>`_.
+  
+- Strive for clarity. Sometimes nonstandard indentation, especially when combined with non-syntactic line breaks, makes things easier to read.
+
+  - Make sure that line breaks and indentation stay within the valid syntax of the language.
+
+Using two spaces keeps code sample lines shorter, which makes them easier to view.
+
+
+.. rubric:: Example of indenting for clarity
+
+.. code-block:: HTTP
+
+  HTTP/1.0 401 Unauthorized
+  Server: HTTPd/0.9
+  Date: Sun, 10 Apr 2005 20:26:47 GMT
+  WWW-Authenticate: Digest realm="testrealm@host.com",
+			   qop="auth,auth-int",
+			   nonce="dcd98b7102dd2f0e8b11d0f600bfb0c093",
+			   opaque="5ccc069c403ebaf9f0171e9517f40e41"
+  Content-Type: text/html
+  Content-Length: 311
+
+.. _meaningful-names:
+
+Meaningful names
+-----------------
+
+When writing sample code, avoid meaningless names.
+
+.. rubric:: Wrong
+
+.. code-block:: python
+
+  def myFunction(foo):
+
+    for bar in foo:
+       bar[foo] = foo[spam] + spam[foo]
+
+    return foobar
+
+.. _xml-html-style-guide:
+
+XML and HTML
+---------------
+
+Some of the terms often used to describe XML and HTML code structures are imprecise or confusing. For clarity, we restrict certain terms and uses.
+
+Likewise, coding practices and styles for XML and HTML vary widely. For the sake of clarity and consistency, samples should follow the guidelines set forth here.
+
+.. _xml-element:
+
+Element
+~~~~~~~~~~~
+
+The following piece of code represents an **element**:
+
+.. code-block:: xml
+
+  <element>
+    Some content.
+  </element>
+
+.. note:: 
+
+  An element is **not** a *block* or a *tag*.
+  
+  - *Tag* is defined below.
+  - *Block* has a specific meaning in HTML and XML templates, and should generally be avoided outside those contexts.
+
+.. _xml-tag:
+  
+Tag
+~~~~~~
+
+A **tag** is the token that begins or ends an element.
+
+.. code-block:: xml
+
+  <element>  <!-- The opening tag of this element. -->
+    Some content.
+  </element> <!-- The closing tag. -->
+  
+The word *tag* has often been used to refer to the entire element. For clarity, we will avoid that here.
+
+
+.. _xml-node:
+
+Node
+~~~~~
+
+The word *node* is often used interchangeably with *element*.
+
+For clarity, we make the following distinction:
+
+- An *element* is a piece of XML or HTML code.
+- A *node* is an element rendered into a DOM tree or other dynamic representation.
```

Comment:
I think this is quite confusing as there are different types of nodes like element node, document node, etc., and a node is a part of DOM tree, this should be the other way around.I guess **An element is a particular kind of node** would be accurate.

Reply:
This is a good point.

The thing I was trying to get across is that some code in a ` *.xml` file is **not** a node. A node is something in a "live" DOM tree. 
Look at my next commit and see if it makes more sense then.